extensions/renderer/resources/guest_view/web_view/web_view.js

```diff
@@ -148,11 +148,12 @@ WebViewImpl.prototype.onFrameNameChanged = function(name) {
 // Updates state upon loadcommit.
 WebViewImpl.prototype.onLoadCommit = function(
     baseUrlForDataUrl, currentEntryIndex, entryCount,
-    processId, url, isTopLevel) {
+    processId, url, isTopLevel, pagesHistory) {
   this.baseUrlForDataUrl = baseUrlForDataUrl;
```

Comment:
I don’t know about limit for IPC message. But it will not grow longer and longer. For navigation history there is some limit. When you have about 30 or so items it will remove the oldest. Or at least it seems to be the case after quick test.